ecleankernel/file.py

```diff
@@ -91,6 +96,42 @@ def __init__(self,
         super().__init__(path, KernelFileType.KERNEL)
         self.internal_version = self.read_internal_version()
 
+    def __decompress_raw(self) -> bytes:
+        f = open(self.path, 'rb')
+        magic_dict = {
+            b'\x1f\x8b\x08': 'gzip',
+            b'\x42\x5a\x68': 'bz2',
+            b'\xfd\x37\x7a\x58\x5a\x00': 'lzma',
+            b'\x5d\x00\x00': 'lzma',
+            b'\x04\x22\x4d\x18': 'lz4.frame',
+            b'\x28\xb5\x2f\xfd': 'zstandard',
+            b'\x89\x4c\x5a\x4f\x00\x0d\x0a\x1a\x0a': 'lzo'
+            }
+        maxlen = max(len(x) for x in magic_dict)
+        header = f.read(maxlen)
+        f.seek(0)
+        for magic, comp in magic_dict.items():
+            if header.startswith(magic):
+                try:
+                    mod = importlib.import_module(comp)
+                except ModuleNotFoundError:
+                    raise MissingDecompressorError(
+                        f'Kernel file {self.path} is compressed with '
+                        f'{comp}, but the required decompressor '
+                        f'is not installed')
+                if comp == 'zstandard':
+                    # Technically a redundant import, this is just
+                    # to make your IDE happy :)
+                    import zstandard
+                    b = zstandard.ZstdDecompressor().decompress(f.read())
+                else:
+                    b = getattr(mod, 'decompress')(f.read())
```

Comment:
```suggestion
                    return getattr(mod, 'decompress')(f.read())
```

You don't really have to iterate over all of them, esp. that magics don't overlap.

Reply:
For extra kudos, you could try seeing if all of them can have some streaming classes (i.e. `open()` function), so we could avoid loading the whole file into memory unnecessarily.

Reply:
I somehow didn't think of just returning them - oops

Sadly, lzo only provides `compress()` and `decompress()` functions :/

Reply:
Ok, I suppose this makes converting others pointless.